storage/remote/write_handler.go

```diff
@@ -479,56 +484,99 @@ func (h *writeHandler) appendV2(app storage.Appender, req *writev2.Request, rs *
 	return samplesWithoutMetadata, http.StatusBadRequest, errors.Join(badRequestErrs...)
 }
 
+type OTLPOptions struct {
+	// Convert delta samples to their cumulative equivalent by aggregating in-memory
+	ConvertDelta bool
+}
+
 // NewOTLPWriteHandler creates a http.Handler that accepts OTLP write requests and
 // writes them to the provided appendable.
-func NewOTLPWriteHandler(logger *slog.Logger, appendable storage.Appendable, configFunc func() config.Config) http.Handler {
-	rwHandler := &writeHandler{
-		logger:     logger,
-		appendable: appendable,
+func NewOTLPWriteHandler(logger *slog.Logger, reg prometheus.Registerer, appendable storage.Appendable, configFunc func() config.Config, opts OTLPOptions) http.Handler {
+	ex := &rwExporter{
+		writeHandler: &writeHandler{
+			logger:     logger,
+			appendable: appendable,
+		},
+		config: configFunc,
+	}
+
+	steps := []otel.Step{otel.Sink(ex)}
+	if opts.ConvertDelta {
+		steps = []otel.Step{
```

Comment:
Maybe silly question, but do we need Prometheus to literally re-create a full Otel collector framework? Can we use `deltatocumulative` directly, before passing to remote write to avoid the cost and complexity of those abstractions? 

If we go this path, we end up with the same flow (with all its cost, reliability and complexity) as running configured OtelCollector sidecar to Prometheus binary (just minus localhost hop + serialization). Do we want that?

This can be trivially optimized by directly invoking core delta to cumulative function, only a delta type metris in a single for-loop. This will be both faster and much simpler, especially to answer question "how much it impacts efficiency for non-delta metrics" or even "how much it impacts efficiency for otel-delta feature flag disabled.